4.gene-expression-signatures/scripts/2.run_gsea.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import joblib
+import pandas as pd
+import blitzgsea as blitz
+import random
+import pathlib
+import sys
+
+script_directory = pathlib.Path("../2.train-VAE/utils/").resolve()
+sys.path.insert(0, str(script_directory))
+from betavae import BetaVAE, weights
+from betatcvae import BetaTCVAE, tc_weights
+from vanillavae import VanillaVAE, vanilla_weights
+
+script_directory = pathlib.Path("../utils/").resolve()
+sys.path.insert(0, str(script_directory))
+from data_loader import load_train_test_data, load_model_data
+
+
+# In[2]:
+
+
+# Load data
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+
+train_df, test_df, val_df, load_gene_stats = load_train_test_data(
+    data_directory, train_or_test="all", load_gene_stats=True, zero_one_normalize=True
+)
+train_data = pd.DataFrame(train_df)
+
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve()
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet").resolve()
+dependency_df, gene_dict_df= load_model_data(dependency_file, gene_dict_file)
+gene_dict_df = pd.DataFrame(gene_dict_df)
+
+
+# In[3]:
+
+
+#Load weight data for VAEs
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+weight_df = load_train_test_data(
+    data_directory, train_or_test="train"
+)
+
+gene_list_passed_qc = gene_dict_df.loc[
+    gene_dict_df["qc_pass"], "dependency_column"
+].tolist()
+
+weight_data = weight_df.filter(gene_list_passed_qc, axis=1)
+weight_data.head()
+
+
+# In[4]:
+
+
+# Function to extract weights for sklearn models
+def extract_weights(model, model_name):
```

Comment:
Consider giving this function type hints and a docstring (possibly including the comment above). Also consider doing the same for the function below.